### PR TITLE
(AB#33081) Update for 7.3.0 release

### PR DESCRIPTION
**Pull request by michaeltlombardi**
# PR Summary

This change updates the documentation to address the latest stable version of PowerShell being `7.3.0`.

- Fixes [AB#33081](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/33081)

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: reference/docs-conceptual/install/Installing-PowerShell-on-Windows.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ winget search Microsoft.PowerShell
 ```Output
 Name               Id                           Version Source
 ---------------------------------------------------------------
-PowerShell         Microsoft.PowerShell         7.2.7.0 winget
+PowerShell         Microsoft.PowerShell         7.3.0.0 winget
 PowerShell Preview Microsoft.PowerShell.Preview 7.3.8.0 winget
```
Comment (sdwheeler) [inline reference/docs-conceptual/install/Installing-PowerShell-on-Windows.md:56]:
```suggestion
PowerShell Preview Microsoft.PowerShell.Preview 7.3.101.0 winget
```

File: reference/docs-conceptual/install/PowerShell-Support-Lifecycle.md [hunk 2]
```diff
@@ -112,18 +112,20 @@ use the traditional, paid support options.
...
 ## PowerShell End-of-support dates
 
-Based on these lifecycle policies, the following table lists the dates when currently supported version will no
-longer be supported:
+Based on these lifecycle policies, the following table lists the dates when currently supported
+version will no longer be supported:
 
 |      Version      |   End-of-support   |
 | ----------------- | ------------------ |
+| 7.3 (Stable)      | - |
```
Comment (sdwheeler) [inline reference/docs-conceptual/install/PowerShell-Support-Lifecycle.md:120]: @SteveL-MSFT mentioned that support for .NET is 18 months so we may need to adjust these dates.

**Review by sdwheeler — DISMISSED** [review]
See my comments

File: reference/docs-conceptual/install/PowerShell-Support-Lifecycle.md (hunk 2 above)
Comment (michaeltlombardi) [reply reference/docs-conceptual/install/PowerShell-Support-Lifecycle.md:120]: That puts 7.3 at May 8, 2024, do we plan to retroactively modify the dates for 7.1 and earlier?

**Pull request merged**